terratorch/tasks/segmentation_tasks.py

```diff
@@ -278,14 +280,17 @@ def validation_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -
         rest = {k: batch[k] for k in other_keys}
         model_output: ModelOutput = self(x, **rest)
         loss = self.val_loss_handler.compute_loss(model_output, y, self.criterion, self.aux_loss)
-        self.val_loss_handler.log_loss(self.log, loss_dict=loss, batch_size=x.shape[0])
+        self.val_loss_handler.log_loss(self.log, loss_dict=loss, batch_size=y.shape[0])
         y_hat_hard = to_segmentation_prediction(model_output)
         self.val_metrics.update(y_hat_hard, y)
 
         if self._do_plot_samples(batch_idx):
             try:
                 datamodule = self.trainer.datamodule
                 batch["prediction"] = y_hat_hard
+                if isinstance(batch["image"], dict):
```

Comment:
@blumenstiel nooby question, but aren't we loosing flexibility by expecting the image key always to be present? For example in the wxc models it is not present. Or are we saying if there is an image present then we unpack it in case it is packed? or are you using image here some sort of equivalent for 'payload'?

Reply:
It's a segmentation task, so the task definitely expects a key `"image"`. It is also required in the code before and in albumentations.
But we cannot assume that `rgb_modality` is always present in the datamodule. I will change that.